### PR TITLE
refactor: move upload c2 code from controller to service(fpla-2186)

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-2186

### Change description ###

move upload c2 code from controller to service

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsService.java [hunk 1]
```diff
@@ -1,21 +1,60 @@
...
+        List<Element<C2DocumentBundle>> c2DocumentBundle = defaultIfNull(caseData.getC2DocumentBundle(),
+            Lists.newArrayList());
+
+        List<SupportingEvidenceBundle> updatedSupportingEvidenceBundle =
+            unwrapElements(caseData.getTemporaryC2Document().getSupportingEvidenceBundle())
+                .stream()
+                .map(supportingEvidence -> supportingEvidence.toBuilder().dateTimeUploaded(time.now()).build())
+                .collect(Collectors.toList());
+
+        var c2DocumentBundleBuilder = caseData.getTemporaryC2Document().toBuilder()
```
Comment (Jamesnelson117) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsService.java:46]: Can we not just type this as `C2DocumentBundle.C2DocumentBundleBuilder`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsService.java [hunk 2]
```diff
@@ -1,21 +1,60 @@
...
+            .author(idamClient.getUserInfo(requestData.authorisation()).getName())
+            .uploadedDateTime(formatLocalDateTimeBaseUsingFormat(time.now(), DATE_TIME))
+            .supportingEvidenceBundle(wrapElements(updatedSupportingEvidenceBundle));
+
+        c2DocumentBundleBuilder.type(caseData.getC2ApplicationType().get("type"));
```
Comment (Jamesnelson117) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsService.java:51]: Does this need to be separate from the builder above?

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsServiceTest.java [hunk 3]
```diff
@@ -29,9 +41,30 @@
...
+
+    @MockBean
+    private RequestData requestData;
+
+    @Test
+    void shouldBuildExpectedC2DocumentBundle() {
+        given(idamClient.getUserInfo(USER_AUTH_TOKEN)).willReturn(UserInfo.builder().name("Emma Taylor").build());
+        given(requestData.authorisation()).willReturn(USER_AUTH_TOKEN);
+
+        List<Element<C2DocumentBundle>> list = service.buildC2DocumentBundle(createCaseDataWithC2DocumentBundle());
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsServiceTest.java:58]: `actualC2DocumentBundle` perhaps?

File: service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsService.java (hunk 2 above)
Comment (KiranVarma02) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/service/UploadC2DocumentsService.java:51]: don't know why its like that... i just extracted out from controller as you probably know!

**Review by Jamesnelson117 — APPROVED** [review]
LGTM

**Pull request merged**